src/TSnePlotView.js

```diff
@@ -84,6 +84,8 @@ class TSnePlotView extends React.Component {
       this._fetchAndSetStateCellClusters(this.props)
     } else if (previousProps.geneId !== this.props.geneId) {
       this._fetchAndSetStateGeneId(this.props)
+    } else if (previousProps.selectedColourBy !== this.props.selectedColourBy) {
```

Comment:
This condition could be combined into 
```
else if (previousProps.selectedColourByCategory !== this.props.selectedColourBy &&
previousProps.selectedColourBy !== this.props.selectedColourBy)
```
where I believe its a bug here, because `previousProps.selectedColourByCategory !== this.props.selectedColourBy ` will always be true @alfonsomunozpomer @monicajianu .

Hence I think the combined condition should be 
```
else if (previousProps.selectedColourByCategory !== this.props.selectedColourByCategory ||
previousProps.selectedColourBy !== this.props.selectedColourBy)
```

Reply:
I agree seems like bug. Have modified the conditional statement accordingly.